specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHubCluster.json

```diff
@@ -0,0 +1,606 @@
+            "schema": {
+              "$ref": "#/definitions/EHCluster"
+            }
+          },
+          "202": {
+            "description": "Event Hubs Cluster update request accepted."
+          },
+          "default": {
+            "description": "Event Hubs error response detailing why the Event Hubs Cluster update operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/clusters/{clusterName}/namespaces": {
+      "get": {
+        "tags": [
+          "Event Hubs Namespaces in Cluster"
+        ],
+        "operationId": "Clusters_NamespacesGet",
+        "x-ms-examples": {
+          "ClusterNamespaces": {
+            "$ref": "./examples/Clusters/EHNamespacesInClusterGet.json"
+          }
+        },
+        "description": "Get all Event Hubs Namespaces within the Event Hubs Cluster.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Event Hubs Cluster Namespace list returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/EHNamespaceListResult"
+            }
+          },
+          "default": {
+            "description": "Event Hubs Cluster error response describing why the namespaces in cluster get operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/clusters/{clusterName}/quotaConfiguration/default": {
+      "patch": {
+        "tags": [
+          "Event Hubs Cluster Configuration"
+        ],
+        "operationId": "Clusters_ConfigurationPatch",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EHClusterQuotaConfigurationProperties",
```

Comment:
Can this be more strongly typed?

Reply:
These configurations are service-level properties that the RP is exposing to the customer through the control plane in this API. We need this to remain a string mapping type so that the service can add additional properties without requiring ARM API version incrementation each time.

Reply:
Is it possible to expose an API to get a list of applicable properties? We would like APis to be as easy to use as possible.